cpp/src/utilities/stream_pool.cpp

```diff
@@ -161,36 +161,45 @@ cuda_stream_pool* create_global_cuda_stream_pool()
   return new rmm_cuda_stream_pool();
 }
 
+// FIXME: these will be available in rmm soon
+inline int get_num_cuda_devices()
+{
+  rmm::cuda_device_id::value_type num_dev{};
+  CUDF_CUDA_TRY(cudaGetDeviceCount(&num_dev));
+  return num_dev;
+}
+
+rmm::cuda_device_id get_current_cuda_device()
+{
+  int device_id;
+  CUDF_CUDA_TRY(cudaGetDevice(&device_id));
+  return rmm::cuda_device_id{device_id};
+}
+
 /**
  * @brief RAII struct to wrap a cuda event and ensure its proper destruction.
  */
 struct cuda_event {
-  cuda_event()
-    : e_{[]() {
-        cudaEvent_t event;
-        CUDF_CUDA_TRY(cudaEventCreateWithFlags(&event, cudaEventDisableTiming));
-        return event;
-      }()}
-  {
-  }
+  cuda_event() { CUDF_CUDA_TRY(cudaEventCreateWithFlags(&e_, cudaEventDisableTiming)); }
+  virtual ~cuda_event() { CUDF_ASSERT_CUDA_SUCCESS(cudaEventDestroy(e_)); }
 
-  operator cudaEvent_t() { return e_.get(); }
+  operator cudaEvent_t() { return e_; }
 
  private:
-  struct deleter {
-    using pointer = cudaEvent_t;
-    auto operator()(cudaEvent_t e) { CUDF_ASSERT_CUDA_SUCCESS(cudaEventDestroy(e)); }
-  };
-  std::unique_ptr<cudaEvent_t, deleter> e_;
+  cudaEvent_t e_;
 };
 
 /**
  * @brief Returns a cudaEvent_t for the current thread.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion:** ‏ I think This will fail the new doxygen tests because the return value is not documented. There is a new `briefreturn` doxygen macro created by @vyasr that you can use to keep the docs a one-liner.

Reply:
Nevermind, the doxygen tests apply to RMM only so far.